src/LibraryManager.Vsix/Shared/LibraryCommandService.cs

```diff
@@ -0,0 +1,456 @@
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TaskStatusCenter;
+using Microsoft.VisualStudio.Telemetry;
+using Microsoft.Web.LibraryManager.Contracts;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Web.LibraryManager.Vsix
+{
+    [Export(typeof(ILibraryCommandService))]
+    internal class LibraryCommandService : ILibraryCommandService, IVsSolutionEvents, IDisposable
+    {
+        [Import(typeof(ITaskStatusCenterService))]
+        internal ITaskStatusCenterService TaskStatusCenterServiceInstance;
+
+        private readonly IVsSolution _solution;
+        private readonly uint _solutionCookie;
+        private CancellationTokenSource _linkedCancellationTokenSource;
+        private CancellationTokenSource _internalCancellationTokenSource;
+        private Task _currentOperationTask;
+        private object _lockObject = new object(); 
+
+        [ImportingConstructor]
+        public LibraryCommandService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            _solution = (IVsSolution)serviceProvider.GetService(typeof(SVsSolution));
+            _solution.AdviseSolutionEvents(this, out _solutionCookie);
+        }
+
+        public bool IsOperationInProgress
+        {
+            get
+            {
+                lock (_lockObject)
+                {
+                    return _currentOperationTask != null && !_currentOperationTask.IsCompleted;
+                }
+            }
+        }
+
+        public async Task RestoreAsync(string configFilePath, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (IsOperationInProgress)
+            {
+                return;
+            }
+
+            try
+            {
+                Dependencies dependencies = Dependencies.FromConfigFile(configFilePath);
+                Manifest manifest = await Manifest.FromFileAsync(configFilePath, dependencies, cancellationToken).ConfigureAwait(false);
+
+                string taskTitle = GetTaskTitle(OperationType.Restore, null);
+                ITaskHandler handler = TaskStatusCenterServiceInstance.CreateTaskHandler(taskTitle);
+                CancellationToken internalToken = RegisterCancellationToken(handler.UserCancellation);
+
+                lock (_lockObject)
+                {
+                    _currentOperationTask = RestoreAsync(new Dictionary<string, Manifest>() { [configFilePath] = manifest }, internalToken);
+                     handler.RegisterTask(_currentOperationTask);
+                }
+
+                await _currentOperationTask.ConfigureAwait(false);
```

Comment:
Could this just be

await RestoreAsync(configFilePath, manifest, cancellationToken);

?

Reply:
Fixed